openmls/src/treesync/node/leaf_node.rs

```diff
@@ -438,7 +487,7 @@ impl LeafNode {
         capabilities: Capabilities,
         extensions: Extensions,
         backend: &impl OpenMlsCryptoProvider,
-    ) -> Result<(Self, EncryptionKeyPair), LibraryError> {
+    ) -> Result<(LeafNode<StateUnvalidated>, EncryptionKeyPair), LibraryError> {
```

Comment:
Not sure about this. I think we should try to ensure that every leaf node we create is valid. Returning `LeafNode<Unvalidated>` here was a quick way to make the compiler happy. Validation will need to be done later (when more appropriate).